### PR TITLE
ES2016: Exponentiation operator and Unicode 8 changes

### PR DESCRIPTION
**Pull request by anba**
- Exponentiation operator:
  - The expected error needs to be updated for one test (e235e91)
- Unicode 8:
  - Updated white space tests for U+180E (d48bc8c)
  - Added new tests to ensure U+180E is not recognized as white space (0a8ac01)

cc @littledan

### CODE REVIEW

**Comment by littledan** [comment]
Yay! +1

**Comment by rwaldron** [comment]
Where are the parts related to Exponentiation operator?

**Comment by anba** [comment]
e235e9125f8bc259935f6d506c40200c8224f508 needs to be removed, `++ UnaryExpression` is allowed (again), which means `++ ++ x` produces an early reference error instead of a syntax error. Older drafts for the exponentiation operator only allowed `++ LeftHandSideExpression`, cf. https://github.com/rwaldron/exponentiation-operator/issues/17.

**Comment by anba** [comment]
Update: Rebased and removed no longer valid commit for exponentiation operator changes.

File: test/built-ins/Number/S9.3.1_A2_U180E.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2009 the Sputnik authors.  All rights reserved.
```
Comment (leobalter) [inline test/built-ins/Number/S9.3.1_A2_U180E.js]: needs to update the year and author. Other new files on this branch have the same issue.

File: test/built-ins/Number/S9.3.1_A2_U180E.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+id: sec-tonumber-applied-to-the-string-type
```
Comment (leobalter) [inline test/built-ins/Number/S9.3.1_A2_U180E.js]: id got replaced by esid

File: test/built-ins/String/prototype/trim/15.5.4.20-3-3_U180E.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2012 Ecma International.  All rights reserved.
```
Comment (leobalter) [inline test/built-ins/String/prototype/trim/15.5.4.20-3-3_U180E.js]: year and author too

**Comment by leobalter** [comment]
@anba ping. I can replace the id tags but I believe it's good to hear from you on the licensing comments above

**Comment by leobalter** [comment]
\o/ pinging @anba again. \o/

**Comment by anba** [comment]
Superseded by #709

**Pull request closed without merging**